gc.c

```diff
@@ -1940,12 +2018,12 @@ object_id(VALUE obj)
 
     if (UNLIKELY(rb_gc_multi_ractor_p() && rb_ractor_shareable_p(obj))) {
```

Comment:
```suggestion
    if (UNLIKELY(rb_gc_multi_ractor_p() && RB_OBJ_SHAREABLE_P(obj))) {
```

Faster and more restrictive check.